backend/src/schema/resolvers/posts.js

```diff
@@ -211,10 +225,75 @@ export default {
       })
       return emoted
     },
+    pinPost: async (_parent, params, context, _resolveInfo) => {
+      let pinnedPostWithNestedAttributes
+      const { driver, user } = context
+      const session = driver.session()
+      const { id: userId } = user
+      let writeTxResultPromise = session.writeTransaction(async transaction => {
+        const deletePreviousRelationsResponse = await transaction.run(
+          `
+          MATCH (:User)-[previousRelations:PINNED]->(post:Post)
+          DELETE previousRelations
```

Comment:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-10-18T13:15:48Z" title="Friday, October 18th 2019, 3:15:48 pm +02:00">Oct 18, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -211,10 +225,68 @@ export default {
       })
       return emoted
     },
+    pinPost: async (_parent, params, context, _resolveInfo) => {
+      let pinnedPost
+      const { driver, user } = context
+      const session = driver.session()
+      const { id: userId } = user
+      let writeTxResultPromise = session.writeTransaction(async transaction => {
+        const deletePreviousRelationsResponse = await transaction.run(
+          `
+          MATCH ()-[previousRelations:PINNED]->(post:Post)
+          DELETE previousRelations
```


What happens with the posts are not pinned anymore?
They appear as usual post?
That would be annoying, or?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-18T16:20:25Z" title="Friday, October 18th 2019, 6:20:25 pm +02:00">Oct 18, 2019</time>_

---


Yes, what should happen with them? They are part of the network... Why should they be disabled or deleted? Or do you mean something else

Reply:
> [<img alt="aonomike" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/aonomike) **Authored by [aonomike](https://github.com/aonomike)**
_<time datetime="2019-10-18T17:17:08Z" title="Friday, October 18th 2019, 7:17:08 pm +02:00">Oct 18, 2019</time>_

---


And how long does it take before an admin can unpin a post? I was thinking of a case that one admin has just pinned a post, and immediately after another admin pins a different post.

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-19T09:43:02Z" title="Saturday, October 19th 2019, 11:43:02 am +02:00">Oct 19, 2019</time>_

---


You can pin another `Post` straight after @aonomike... We shouldn't have an issue like you described cause we have very few admins that will pin posts, and hopefully will not abuse pinning them.
If we do have an issue, we can address it.